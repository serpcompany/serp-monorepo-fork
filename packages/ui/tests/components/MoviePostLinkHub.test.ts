/* eslint-disable @typescript-eslint/no-unused-vars, no-unused-vars  */

import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import MoviePostLinkHub from '../../components/MoviePostLinkHub.vue';
import ComponentRender from '../componentRender';

// Mock data variable to be updated in each test case
let postsData_: Array<{
  slug: string;
  keyword: string;
  featuredImage: string;
}> = [];

// Mock the usePosts composable
mockNuxtImport('usePosts', () => async (...args: unknown) => ({
  posts: postsData_
}));

describe('MoviePostLinkHub Snapshot', () => {
  const scenarios: [
    string,
    { posts: Array<{ slug: string; keyword: string; featuredImage: string }> }
  ][] = [
    [
      'with posts',
      {
        posts: [
          {
            slug: 'movie-1',
            keyword: 'Movie One',
            featuredImage: 'https://example.com/movie1.jpg'
          },
          {
            slug: 'movie-2',
            keyword: 'Movie Two',
            featuredImage: 'https://example.com/movie2.jpg'
          }
        ]
      }
    ],
    ['with no posts', { posts: [] }]
  ];

  it.each(scenarios)('renders %s correctly', async (desc, { posts }) => {
    // Update the posts data for this test case
    postsData_ = posts;

    const html = await ComponentRender(
      `MoviePostLinkHub ${desc}`,
      {},
      MoviePostLinkHub
    );
    expect(html).toMatchSnapshot();
  });
});
