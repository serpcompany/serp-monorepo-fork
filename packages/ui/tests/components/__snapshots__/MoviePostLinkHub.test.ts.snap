// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoviePostLinkHub Snapshot > renders with no posts correctly 1`] = `"<div role="list" class="relative flex flex-col *:not-last:after:absolute *:not-last:after:inset-x-1 *:not-last:after:bottom-0 *:not-last:after:bg-border *:not-last:after:h-px"></div>"`;

exports[`MoviePostLinkHub Snapshot > renders with posts correctly 1`] = `
"<div role="list" class="relative flex flex-col *:not-last:after:absolute *:not-last:after:inset-x-1 *:not-last:after:bottom-0 *:not-last:after:bg-border *:not-last:after:h-px">
  <div data-orientation="vertical" class="relative flex rounded-lg transition hover:bg-elevated/50">
    <!--v-if-->
    <div class="relative flex flex-col flex-1 lg:grid gap-x-8 gap-y-4 p-4 sm:p-6">
      <div class="flex flex-col flex-1 items-start">
        <!--v-if-->
        <!--v-if-->
        <div class="flex-1">
          <div data-orientation="horizontal" class="relative group/user flex items-center gap-2.5"><span class="inline-flex items-center justify-center select-none overflow-hidden rounded-full align-middle bg-elevated size-10 text-xl shrink-0"><img width="40" height="40" alt="Movie One" data-nuxt-img="" srcset="https://example.com/movie1.jpg 1x, https://example.com/movie1.jpg 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://example.com/movie1.jpg"></span>
            <div class="">
              <!--v-if-->
              <p class="font-medium text-highlighted text-base">Movie One</p>
              <!--v-if-->
            </div>
          </div>
        </div>
        <!--v-if-->
      </div>
    </div><a href="/movies/movie-1" tabindex="-1" aria-label="Card link" class="focus:outline-none peer"><span class="absolute inset-0" aria-hidden="true"></span></a>
  </div>
  <div data-orientation="vertical" class="relative flex rounded-lg transition hover:bg-elevated/50">
    <!--v-if-->
    <div class="relative flex flex-col flex-1 lg:grid gap-x-8 gap-y-4 p-4 sm:p-6">
      <div class="flex flex-col flex-1 items-start">
        <!--v-if-->
        <!--v-if-->
        <div class="flex-1">
          <div data-orientation="horizontal" class="relative group/user flex items-center gap-2.5"><span class="inline-flex items-center justify-center select-none overflow-hidden rounded-full align-middle bg-elevated size-10 text-xl shrink-0"><img width="40" height="40" alt="Movie Two" data-nuxt-img="" srcset="https://example.com/movie2.jpg 1x, https://example.com/movie2.jpg 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://example.com/movie2.jpg"></span>
            <div class="">
              <!--v-if-->
              <p class="font-medium text-highlighted text-base">Movie Two</p>
              <!--v-if-->
            </div>
          </div>
        </div>
        <!--v-if-->
      </div>
    </div><a href="/movies/movie-2" tabindex="-1" aria-label="Card link" class="focus:outline-none peer"><span class="absolute inset-0" aria-hidden="true"></span></a>
  </div>
</div>"
`;
